tools/ci_build/github/apple/build_ios_framework.py

```diff
@@ -72,7 +72,9 @@ def _build_package(args):
 
         # get the compiled lib path
         framework_dir = os.path.join(
-            build_dir_current_arch, build_config, build_config + "-" + sysroot, 'onnxruntime.framework')
+            build_dir_current_arch, build_config, build_config + "-" + sysroot,
+            '' if args.build_dynamic_framework else 'static_framework',
```

Comment:
>'' [](http://example.com/codeflow?start=12&length=2)

how does os.path.join() handle an empty path component?

Reply:
os.path.join() will ignore empty path component, (at least on MacOS), changed it to a better way
As for a separated path for dynamic_framework, can do it later, since I want to keep this change minimal for cherry picking